WordPress/openverse
# Update renovate config to handle all dependency types and group upgrades

## Fixes



Fixes #3304 by @sarayourfriend
Closes https://github.com/WordPress/openverse-infrastructure/issues/484 by @sarayourfriend

## Description


Updates the existing renovate configuration in the following ways:

- Match dependabot stack and tech labeling
- Group dependencies that can be grouped (note: we can continue iterating on this, there are probably lots of good opportunities for this in the frontend especially, I just picked some obvious ones)
- Update configuration options that were in an older format
- Add a pre-commit hook to validate the renovate configuration
- Remove the dependabot configuration (also requires disabling it in the repo settings, I think)

**This is _one_ approach we could take**. The other is, of course, to continue using dependabot. It has rudimentary support for pnpm and grouped dependencies, both added very recently, but it's a bit more of a chore to configure (lots of repeated labeling rules, for example) and tends to take a very long time to support new features (e.g., pnpm, grouped dependencies, sensible github action matching), if it ever gets them at all.

## Testing Instructions

Not entirely sure. I suppose we can run the renovate workflow from this branch and see what happens.

## Checklist


- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
